resource_hub/lib/Pages/Bookmarks.dart

```diff
@@ -108,7 +119,46 @@ class _BookmarksState extends State<Bookmarks> {
 
     return;
   }
+  Future<dynamic> pressLike(int post_id, int index) async{
+    if(likedPosts.contains(post_id)){
+      likedPosts.remove(post_id);
+      posts[index]['Likes']--;
+      setState(() {
+
+      });
+      Firestore.instance.collection("Users").where('UserID',isEqualTo: '1').getDocuments().then((data){
+        print("Data is: ${data.documents[0].documentID}");
+
+        var list = List<int>();
+        list.add(post_id);
+        Firestore.instance.collection('Users').document(data.documents[0].documentID).updateData({"Likes": FieldValue.arrayRemove(list)});
+      });
+      Firestore.instance.collection("Posts").where('post_id',isEqualTo: post_id).getDocuments().then((data){
+        print("Data is: ${data.documents[0].documentID}");
+        Firestore.instance.collection('Posts').document(data.documents[0].documentID).updateData({"Likes": posts[index]['Likes']});
+      });
+    }
+    else{
+      setState(() {
+
+      });
```

Comment:
Having empty set state will reload (re-build) the whole widget tree. It will have performance issues later on.

Reply:
what are the alternatives?